roles/container-engine/docker/tasks/systemd.yml

```diff
@@ -26,6 +26,15 @@
   notify: restart docker
   when: not (ansible_os_family in ["CoreOS", "Coreos", "Container Linux by CoreOS"] or is_atomic)
 
+- name: Detect init system used
+  shell: ps -p 1 -o comm=
+  register: init_detection
+
+- name: Detect if systemd is used
+  set_fact:
+    ansible_system_mgr: systemd
+  when: init_detection.rc == 0 and init_detection.stdout == "systemd"
```

Comment:
May be should use ansible facts variable ?
ansible_service_mgr == 'systemd'

Reply:
@LuckySB , I had that same thought but I noticed in the cluster.yml playbook that facts were turned off. I wasn't sure if there was a reason for that, which is why I did this.

Reply:
gather_facts disabled only for some first plays. 
container-engine roles plays with facts